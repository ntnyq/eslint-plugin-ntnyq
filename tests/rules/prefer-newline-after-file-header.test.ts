import { expect } from 'vitest'
import rule, { RULE_NAME } from '../../src/rules/prefer-newline-after-file-header'
import { $, run } from '../internal'

run({
  name: RULE_NAME,
  rule,
  valid: [
    {
      description: 'no-file-header',
      filename: 'no-file-header.ts',
      code: $`
        export const foobar = 'foobar'
      `,
    },
    {
      description: 'no-node',
      filename: 'no-node.ts',
      code: $`
        /**
         * @file eslint config
         */
      `,
    },
    {
      description: 'not-start-header',
      filename: 'not-start-header.ts',
      code: $`
           /**
        * @file eslint config
        */
      `,
    },
    {
      description: 'not-top-header',
      filename: 'not-top-header.ts',
      code: $`
        export const foobar = 'foobar'
        
        /**
        * @file eslint config
        */
      `,
    },
    {
      description: 'line-comment',
      filename: 'line-comment.ts',
      code: $`
        // file eslint config
        export const foobar = 'foobar'
      `,
    },
    {
      description: 'one-newline',
      filename: 'one-newline.ts',
      code: $`
        /**
         * @file eslint config
         */
        
        export const foobar = 'foobar'
      `,
    },
    {
      description: 'not-file-header-jsdoc',
      filename: 'not-file-header-jsdoc.ts',
      code: $`
        /**
         * @foobar eslint config
         */
        
        export const foobar = 'foobar'
      `,
    },
    {
      description: 'multi-newline',
      filename: 'multi-newline.ts',
      code: $`
        /**
         * @file eslint config
         */
        
        
        
        export const foobar = 'foobar'
      `,
    },
  ],
  invalid: [
    {
      description: 'no-newline',
      filename: 'no-newline.ts',
      code: $`
        /**
         * @file eslint config
         */
        export const foobar = 'foobar'
      `,
      output(output) {
        expect(output).toMatchSnapshot('output')
      },
      errors(errors) {
        expect(errors).toMatchSnapshot('errors')
      },
    },
    {
      description: 'same-line',
      filename: 'same-line.ts',
      code: $`
        /**
         * @file eslint config
         */export const foobar = 'foobar'
      `,
      output(output) {
        expect(output).toMatchSnapshot('output')
      },
      errors(errors) {
        expect(errors).toMatchSnapshot('errors')
      },
    },
    {
      description: 'user-defined-jsdoc',
      filename: 'user-defined-jsdoc.ts',
      options: {
        tags: ['@foobar'],
      },
      code: $`
        /**
         * @foobar eslint config
         */export const foobar = 'foobar'
      `,
      output(output) {
        expect(output).toMatchSnapshot('output')
      },
      errors(errors) {
        expect(errors).toMatchSnapshot('errors')
      },
    },
  ],
})
