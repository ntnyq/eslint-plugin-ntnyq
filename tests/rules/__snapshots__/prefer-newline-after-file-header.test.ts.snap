// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`prefer-newline-after-file-header > invalid > Invalid #0: no-newline > errors 1`] = `
[
  {
    "column": 1,
    "endColumn": 31,
    "endLine": 4,
    "fix": {
      "range": [
        30,
        30,
      ],
      "text": "
",
    },
    "line": 4,
    "message": "require a newline before",
    "messageId": "requireNewlineBefore",
    "nodeType": "ExportNamedDeclaration",
    "ruleId": "prefer-newline-after-file-header",
    "severity": 2,
  },
]
`;

exports[`prefer-newline-after-file-header > invalid > Invalid #0: no-newline > output 1`] = `
"/**
 * @file eslint config
 */

export const foobar = 'foobar'"
`;

exports[`prefer-newline-after-file-header > invalid > Invalid #1: same-line > errors 1`] = `
[
  {
    "column": 4,
    "endColumn": 34,
    "endLine": 3,
    "fix": {
      "range": [
        30,
        30,
      ],
      "text": "

",
    },
    "line": 3,
    "message": "require a newline before",
    "messageId": "requireNewlineBefore",
    "nodeType": "ExportNamedDeclaration",
    "ruleId": "prefer-newline-after-file-header",
    "severity": 2,
  },
]
`;

exports[`prefer-newline-after-file-header > invalid > Invalid #1: same-line > output 1`] = `
"/**
 * @file eslint config
 */

export const foobar = 'foobar'"
`;

exports[`prefer-newline-after-file-header > invalid > Invalid #2: user-defined-jsdoc > errors 1`] = `
[
  {
    "column": 4,
    "endColumn": 34,
    "endLine": 3,
    "fix": {
      "range": [
        32,
        32,
      ],
      "text": "

",
    },
    "line": 3,
    "message": "require a newline before",
    "messageId": "requireNewlineBefore",
    "nodeType": "ExportNamedDeclaration",
    "ruleId": "prefer-newline-after-file-header",
    "severity": 2,
  },
]
`;

exports[`prefer-newline-after-file-header > invalid > Invalid #2: user-defined-jsdoc > output 1`] = `
"/**
 * @foobar eslint config
 */

export const foobar = 'foobar'"
`;
